jme3-plugins/src/gltf/java/com/jme3/scene/plugins/gltf/TextureTransformExtensionLoader.java

```diff
@@ -0,0 +1,146 @@
+ * NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.jme3.scene.plugins.gltf;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.jme3.asset.AssetLoadException;
+import com.jme3.math.Matrix3f;
+import com.jme3.math.Vector3f;
+import com.jme3.scene.Geometry;
+import com.jme3.scene.Mesh;
+import com.jme3.scene.VertexBuffer;
+import com.jme3.texture.Texture2D;
+import java.io.IOException;
+import java.nio.FloatBuffer;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * Extension loader for KHR_texture_transform, which allows for
+ * UV coordinates to be scaled/rotated/translated based on 
+ * transformation properties from textures in the glTF model.
+ *
+ * See spec at https://github.com/KhronosGroup/glTF/tree/main/extensions/2.0/Khronos/KHR_texture_transform
+ *
+ * @author manuelrmo - Created on 11/20/2022
+ */
+public class TextureTransformExtensionLoader implements ExtensionLoader {
+    
+    private final static Logger logger = Logger.getLogger(TextureTransformExtensionLoader.class.getName());
+    
+    private Geometry geomLast = null; // Last geometry created by the GltfLoader.class object
+    private Matrix3f mInvLast = null; // Last transformation matrix (inverted) that was applied
+        
+    /**
+     * Scale/rotate/translate UV coordinates based on a transformation matrix.
+     * Code adapted from scaleTextureCoordinates(Vector2f) in jme3-core/src/main/java/com/jme3/scene/Mesh.java
+     * @param mesh The mesh holding the UV coordinates
+     * @param m The matrix containing the scale/rotate/translate transformations
+     */    
+    private void uvTransform(Mesh mesh, Matrix3f m) {
+        VertexBuffer tc = mesh.getBuffer(VertexBuffer.Type.TexCoord);
+        if (tc == null) {
+            throw new IllegalStateException("The mesh has no texture coordinates");
+        }
+        if (tc.getFormat() != VertexBuffer.Format.Float) {
+            throw new UnsupportedOperationException("Only float texture coord format is supported");
+        }
+        if (tc.getNumComponents() != 2) {
+            throw new UnsupportedOperationException("Only 2D texture coords are supported");
+        }
+        FloatBuffer fb = (FloatBuffer) tc.getData();
+        fb.clear();
+        for (int i = 0; i < fb.limit() / 2; i++) {
+            float x = fb.get();
+            float y = fb.get();
+            fb.position(fb.position() - 2);
+            Vector3f v = m.mult(new Vector3f(x, y, 1));
+            fb.put(v.getX()).put(v.getY());
+        }
+        fb.clear();
+        tc.updateData(fb);        
+    }
+    
+    @Override
+    public Object handleExtension(GltfLoader loader, String parentName, JsonElement parent, JsonElement extension, Object input) throws IOException {
+        if (input instanceof Texture2D) {

```

Comment:
Regarding `if (input instanceof Texture2D)` 
Can input be other than Texture2D?

I am thinking if the `input instanceof Mesh` is called before the `input instanceof Texture2D` then you may just grab the mesh instance there and cache it, this way you can get rid of the `GltfLoader.getLastCreatedGeom()`.

Reply:
Thanks for catching those issues and the insightful comments.

Indeed, the extension loader got corrupted in multithreaded loading tests. 

I've implemented a thread-safe version that fixes those issues, and uses the GltfLoader cache as per your advice.